vxyowsup/tests/test_whatsapp.py

```diff
@@ -120,6 +136,17 @@ def receive(self, data):
         '''
         self.toUpper(data)
 
+    def send_ack(self, node):
+        # TODO: rather send IncomingAckProtocolEntity and test extra fields
+        ack = AckProtocolEntity(_id=node['id'], _class='message')
+        self.receive(ack.toProtocolTreeNode())
+
+    def send_receipt(self, node):
+        # msg_class=None defualt indicates "delivered"
+        # alt: msg_class='read'
+        receipt = IncomingReceiptProtocolEntity(_id=node['id'], _from=node['to'], timestamp=node['id'].split('-')[0], type='read')
```

Comment:
I assume if the timestamp was meant to indicate the creation time of the receipt, then that would be taken care of in the `__init__` method of `IncomingReceiptProtocolEntity`